### ISSUE TITLE
Trigger bullets on signal?

### ISSUE THREAD

**Issue by vexxvakan** [Moonzel/Godot-PerfBullets#8]
Hey @Moonzel!

How would I go about adding a shoot on click function?

right now my gun rotates around the player and has a Marker2D at the muzzle position

![image](https://github.com/Moonzel/Godot-PerfBullets/assets/7051136/2c358005-7293-4d69-88aa-0398cb702c25)

i then created a node2d which simply sets its position to mouse on physics process and use this as the PlayerTracker to aim towards my mouse. Now to my issue: I am currently recreating the PatternManager and the Spawner each time I click my attack button, I then move the spawner to the WeaponAttackPivot global_position and add first the spawner then the PatternManager to the scene. the bullet appears and then I queue free the PatternManager.

![image](https://github.com/Moonzel/Godot-PerfBullets/assets/7051136/c985adba-7244-4e66-8478-5099b3b31c59)

however this way the node id cant fire in quick succession and i run into several race conditions. Is there an easier way to start the pattern, shoot it once (number of shots is at 1) and wait until the player clicks again? Been sitting at it for several hours now and I don't think where I am going is the right way...

**Comment by Moonzel** [comment]
I do not believe that this is the best way to do this, as Spawners being created in this way could potentially lead to some issues with Godot's Physics system. I am in the process of adding a way to start a pattern by emitting a signal, and will comment here again when I have finished. Thanks for letting me know about this problem!

**Comment by Moonzel** [comment]
@vexxvakan The Update 1.2 is now live! I hope you enjoy the new MANUAL mode. This should be just what you needed!

**Comment by vexxvakan** [comment]
> @vexxvakan The Update 1.2 is now live! I hope you enjoy the new MANUAL mode. This should be just what you needed!

@Moonzel Amazing news, let me try the update today and I'll add some feedback later!!
